Pre-release activities (#1021)
insightsengineering/teal: vignettes/getting-started-with-teal.Rmd

  * In the example code: the filter panel is being initialized with a filter for the `Species` variable in the `iris` dataset.

## Creating your own applications

The key function to use to create your `teal` application is `init`, which requires two mandatory arguments: `data` and `modules`. There are other optional arguments for `init`, which can be used to customize the application. Please refer to the documentation for `init` for further details.

### Application data

The `data` argument in the `init` function specifies the data used in your application. All datasets which are about to be used in `teal` application must be passed through `teal_data` object.
It is also possible to specify relationships between the datasets using the `join_keys` argument but in this case the datasets are not related. See [this vignette](including-data-in-teal-applications.html) for details.
If data is not available and has to be pulled from a remote source, `init` must receive a `teal_data_module` that specifies how to obtain the desired datasets and put them into a `teal_data` object.
See [this vignette](data-as-shiny-module.html) for details.

> In order to use CDISC clinical trial data in a `teal` application the `cdisc_data` function is used instead. If you would like to use custom `SDTM` standards, you can use the `teal_data` along with `join_keys` to specify the relationships between the datasets.

Reviewer: ```suggestion
In order to use CDISC clinical trial data in a `teal` application the `cdisc_data` function is used instead. 
Other standards, like SDTM, can be handled by applying `teal_data` and `join_keys`.
```

Reviewer 2: This is misleading. `SDTM` and CDISC are somewhat analogous, they could not be called other standards.
CDISC is the organization that sets the standards. SDTM is the standard set by this organization. CDISC also provides guidelines to create custom SDTM datasets which is why sponsors (pharma companies) can create their own SDTM datasets. I would prefer the previous document as is.

For further details, we recommend exploring the [`teal.data`](https://insightsengineering.github.io/teal.data/) package documentation.

### Modules

> The `modules` argument to `init` consists of a list of `teal` modules (which can be wrapped together using the function `modules`). Core `teal` developers created several universal `teal` modules which can be helpful for any `teal` application. To learn how to create own modules please explore [Creating Custom Modules vignette](creating-custom-modules.html). To use our predefined modules see the references below for links to these modules.

Reviewer: ```suggestion
The `modules` argument to `init` consists of a list of `teal` modules (which can be wrapped together using the function `modules`).
Core `teal` developers have created several universal `teal` modules that can be useful in any `teal` application.
To learn how to create your own modules, please explore [Creating Custom Modules vignette](creating-custom-modules.html).
To use our predefined modules, see the references below for links to these modules.
```

### Defining filters

The optional `filter` argument in `init` allows you to initialize the application with predefined filters.
For further details see [Filter Panel vignette](filter-panel.html) .

### Reporting

If any of the `modules` in your `teal` application support reporting (see [`teal.reporter`](https://insightsengineering.github.io/teal.reporter/) for more details), users of your application can add the outputs of the modules to a report.
This report can then be downloaded and a special _Report Previewer_ module will be added to your application as an additional tab, where users can view and configure their reports before downloading them. See more details in [this vignette](adding-support-for-reporting.html).

### Reproducible code

> `teal` hands over data with reproducible code to every module included in the application. `teal` doesn't display the code it is module's responsibility to render it. For example, the `example_module` function used above shows the code
> in the main panel together with other outputs. For more details see [this vignette](reproducible-code.html).

Reviewer: ```suggestion
`teal` hands over data with reproducible code to every module included in the application.
Note that `teal` does not display the code, that is the modules' responsibility.
For example, the `example_module` function used above shows the code in the main panel together with other outputs.
For more details see [this vignette](reproducible-code.html).
```
